### PR TITLE
Add a secondary control plane LB

### PR DESCRIPTION
**Pull request by nrb**
Current issues:

* Fuzzy conversion tests aren't passing
* AWS Machine controller needs to account for both LBs
* Webhook validation tests will fail because both new and old LBSpecs are `nil`

### CODE REVIEW

File: api/v1beta2/awscluster_webhook.go [hunk 1]
```diff
@@ -243,19 +251,37 @@ func (r *AWSCluster) validateNetwork() field.ErrorList {
...
+func (r *AWSCluster) validateControlPlaneLBs() field.ErrorList {
 	var allErrs field.ErrorList
 
-	if r.Spec.ControlPlaneLoadBalancer == nil {
-		return allErrs
+	// If the secondary is defined, check that the name is not empty and different from the primary.
+	if r.Spec.SecondaryControlPlaneLoadBalancer != nil {
+		if r.Spec.SecondaryControlPlaneLoadBalancer.Name == nil || *r.Spec.SecondaryControlPlaneLoadBalancer.Name == "" {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "secondaryControlPlaneLoadBalancer", "name"), r.Spec.SecondaryControlPlaneLoadBalancer.Name, "secondary controlPlaneLoadBalancer.name cannot be empty"))
+		}
```
Comment (JoelSpeed) [inline api/v1beta2/awscluster_webhook.go:268]: I'm guessing CAPI isn't using CEL validations for their API yet. This is an easy 1 liner in CEL on the secondary field
```
// +kubebuilder:validation:XValidation:rule="has(self.name) && self.name != ''",message="name is required for secondaryControlPlaneLoadBalancer"
```

File: api/v1beta1/awscluster_conversion.go [hunk 2]
```diff
@@ -134,6 +150,10 @@ func restoreControlPlaneLoadBalancer(restored, dst *infrav2.AWSLoadBalancerSpec)
 	dst.DisableHostsRewrite = restored.DisableHostsRewrite
 	dst.PreserveClientIP = restored.PreserveClientIP
 	dst.IngressRules = restored.IngressRules
+	dst.AdditionalSecurityGroups = restored.AdditionalSecurityGroups
+	dst.Scheme = restored.Scheme
+	dst.CrossZoneLoadBalancing = restored.CrossZoneLoadBalancing
+	dst.Subnets = restored.Subnets
```
Comment (nrb) [inline api/v1beta1/awscluster_conversion.go:159]: Adding these fields caused conversion tests to pass, but it seems incorrect if they were missing for the control plane before.

File: api/v1beta2/network_types.go [hunk 3]
```diff
@@ -37,6 +37,9 @@ type NetworkStatus struct {
 	// APIServerELB is the Kubernetes api server load balancer.
 	APIServerELB LoadBalancer `json:"apiServerElb,omitempty"`
 
+	// SecondaryAPIServerELB is the secondary Kubernetes api server load balancer.
+	SecondaryAPIServerELB LoadBalancer `json:"secondaryAPIServerELB,omitempty"`
```
Comment (nrb) [inline api/v1beta2/network_types.go:41]: Does this need to be present? I assume if we create a resource, we want to track it.

Comment (vincepri) [reply api/v1beta2/network_types.go:41]: Yes, makes sense to me to reflect in status

**Pull request closed without merging**